x-pack/plugins/security_solution/public/app/app.tsx

```diff
@@ -110,23 +111,49 @@ const SecurityAppComponent: React.FC<SecurityAppComponentProps> = ({
   setHeaderActionMenu,
   store,
   theme$,
-}) => (
-  <KibanaContextProvider
-    services={{
-      appName: APP_NAME,
-      ...services,
-    }}
-  >
-    <StartApp
-      history={history}
-      onAppLeave={onAppLeave}
-      setHeaderActionMenu={setHeaderActionMenu}
-      store={store}
-      theme$={theme$}
-    >
-      {children}
-    </StartApp>
-  </KibanaContextProvider>
-);
+  isDev,
+}) => {
+  if (isDev) {
+    return (
+      <StrictMode>
+        <KibanaContextProvider
+          services={{
+            appName: APP_NAME,
+            ...services,
+          }}
+        >
+          <StartApp
+            history={history}
+            onAppLeave={onAppLeave}
+            setHeaderActionMenu={setHeaderActionMenu}
+            store={store}
+            theme$={theme$}
+          >
+            {children}
+          </StartApp>
+        </KibanaContextProvider>
+      </StrictMode>
+    );
+  } else {
+    return (
+      <KibanaContextProvider
+        services={{
+          appName: APP_NAME,
+          ...services,
+        }}
+      >
+        <StartApp
+          history={history}
+          onAppLeave={onAppLeave}
+          setHeaderActionMenu={setHeaderActionMenu}
+          store={store}
+          theme$={theme$}
+        >
+          {children}
+        </StartApp>
+      </KibanaContextProvider>
+    );
+  }
+};
```

Comment:
Instead of duplicating the tree, you could create a Wrapper like this:

```jsx
const Wrapper = isDev ? StrictMode : React.Fragment;
```

and then wrap the tree in the `<Wrapper />`:

```jsx
<Wrapper>
  <KibanaContextProvider>...
</Wrapper>
```

Wdyt?